tests/feature/controllers/user.test.js

```diff
@@ -0,0 +1,258 @@
+import User from '../../../src/db/models/user';
+import { UserNotFoundError } from '../../../src/exceptions/errors';
+import * as AuthService from '../../../src/services/auth';
+
+
+const supertest = require('supertest');
+
+const { app } = require('../../../src/app');
+
+// Create our Test AuthToken
+const authToken = AuthService.createToken({
+    id: '6f7a7188-5b7d-4dbb-8a43-12031b94971e',
+    login: 'John Doe',
+    password: 'secret',
+    age: 21,
+    isDeleted: false
+});
+
+describe('Test the users controller', () => {
+    test('It should return all users', async () => {
+        // Mock Model findAll
+        jest.spyOn(User, 'findAll').mockImplementation(() => {
```

Comment:
Consider extracting all the `jest.spyOn`s mocking logic into a utils file a the `tests/` level.
You can write some wrapper function around `jest.spyOn` that will expect a method call and a returned value as arguments. That way each call to that wrapper function will have its on stuff mocked in closure.

Reply:
And drastically reduce repetition of mocking logic :)